overlord/snapstate/snapmgr.go

```diff
@@ -1067,6 +1068,56 @@ func (m *SnapManager) localInstallCleanup() error {
 	return osutil.UnlinkManyAt(d, filenames)
 }
 
+func (m *SnapManager) ensureMountUpdated() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
```

Comment:
Do you want the function to be in snapstate, but be called from servicestate? It was originally in servicestate, but you said previously:

> no strong opinion, but conceptually this belongs to snapstate (as it is the manager that creates the units to start with) and not here

Reply:
no, I asked to move it, this is the right place. What I was trying to say is that this new method should follow more of the patterns of ServiceManager.ensureSnapServicesUpdated (having a flag on the manager, having a way to mock the flag state) to avoid some of the issues we have here atm

Reply:
Done.